File: Participacipacion_4/MiEntero.py
class MiEntero:
    def __init__(self, valor):
        self.valor = valor

    # Devuelve el valor
    def get_valor(self):
        return self.valor

    # Métodos de instancia
    def es_par(self):
        return self.valor % 2 == 0

    def es_impar(self):
        return self.valor % 2 != 0

    def es_primo(self):
        if self.valor <= 1:
            return False
        for i in range(2, self.valor // 2 + 1):
            if self.valor % i == 0:
                return False
        return True

    # Métodos estáticos con int
    @staticmethod
    def es_par_int(n):
        return n % 2 == 0

    @staticmethod
    def es_impar_int(n):
        return n % 2 != 0

    @staticmethod
    def es_primo_int(n):
        if n <= 1:
            return False
        for i in range(2, n // 2 + 1):
            if n % i == 0:
                return False
        return True

    # Métodos estáticos con MiEntero
    @staticmethod
    def es_par_obj(obj):
        return obj.get_valor() % 2 == 0

    @staticmethod
    def es_impar_obj(obj):
        return obj.get_valor() % 2 != 0

    @staticmethod
    def es_primo_obj(obj):
        return obj.es_primo()

    # Métodos equals
    def equals_int(self, n):
        return self.valor == n

    def equals_obj(self, obj):
        return self.valor == obj.get_valor()

    # Métodos parse
    @staticmethod
    def parse_char_array(arr):
        return int("".join(arr))

    @staticmethod
    def parse_string(s):
        return int(s)

File: Participacipacion_4/MiEntero.java
package Participacipacion_4;
public class MiEntero {
    private int valor;

    // Constructor
    public MiEntero(int valor) {
        this.valor = valor;
    }

    // Devuelve el valor
    public int getValor() {
        return valor;
    }

    // Métodos de instancia
    public boolean esPar() {
        return valor % 2 == 0;
    }

    public boolean esImpar() {
        return valor % 2 != 0;
    }

    public boolean esPrimo() {
        if (valor <= 1) return false;
        for (int i = 2; i <= valor / 2; i++) {
            if (valor % i == 0) return false;
        }
        return true;
    }

    // Métodos estáticos con int
    public static boolean esPar(int n) {
        return n % 2 == 0;
    }

    public static boolean esImpar(int n) {
        return n % 2 != 0;
    }

    public static boolean esPrimo(int n) {
        if (n <= 1) return false;
        for (int i = 2; i <= n / 2; i++) {
            if (n % i == 0) return false;
        }
        return true;
    }

    // Métodos estáticos con MiEntero
    public static boolean esPar(MiEntero obj) {
        return obj.getValor() % 2 == 0;
    }

    public static boolean esImpar(MiEntero obj) {
        return obj.getValor() % 2 != 0;
    }

    public static boolean esPrimo(MiEntero obj) {
        return obj.esPrimo();
    }

    // Métodos equals
    public boolean equals(int n) {
        return valor == n;
    }

    public boolean equals(MiEntero obj) {
        return valor == obj.getValor();
    }

    // Métodos parse
    public static int parseInt(char[] arr) {
        return Integer.parseInt(new String(arr));
    }

    public static int parseInt(String s) {
        return Integer.parseInt(s);
    }
}